pkg/kernel/networkservice/connectioncontextkernel/ipcontext/iprule/common.go

```diff
@@ -295,13 +298,23 @@ func del(ctx context.Context, conn *networkservice.Connection, tableIDs *generic
 			if err != nil {
 				return err
 			}
-			for tableID := range ps {
+
+			for tableID, policy := range ps.policymap {
 				nsrtid := NewNetnsRTableNextID(netNS.UniqueId(), tableID)
 				nsRTableNextIDToConnID.Delete(*nsrtid)
-			}
-			for tableID, policy := range ps {
-				if err := delRule(ctx, netlinkHandle, policy, tableID, l.Attrs().Index); err != nil {
-					return err
+				origTID := tableID
```

Comment:
[minor, typo] Probably "nsrtid" and "origTID" have the same meaning of the "tid" suffix. So using the same case lower or upper seems more proper to me